src/AutoRest.CSharp/Mgmt/Generation/NonLongRunningOperationWriter.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Mgmt.Output;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager.Core;
+
+namespace AutoRest.CSharp.Mgmt.Generation
+{
+    internal class NonLongRunningOperationWriter
+    {
+        /// <summary>
+        /// Write a management plane non-LRO as a LRO for the consistency of API surface.
+        /// </summary>
+        /// <param name="writer"></param>
+        /// <param name="operation"></param>
+        public static void Write(CodeWriter writer, NonLongRunningOperation operation)
+        {
+            var responseVariable = "response";
+
+            var cs = operation.Type;
+            var @namespace = cs.Namespace;
+            using (writer.Namespace(@namespace))
+            {
+                writer.WriteXmlDocumentationSummary(operation.Description);
+                var baseType = operation.ResultType != null ? new CSharpType(typeof(Operation<>), operation.ResultType) : new CSharpType(typeof(Azure.Operation));
+                var valueTaskType = operation.ResultType != null ? new CSharpType(typeof(Response<>), operation.ResultType) : new CSharpType(typeof(Response));
+                var waitForCompletionType = new CSharpType(typeof(ValueTask<>), valueTaskType);
+                var helperType = operation.ResultType != null ? new CSharpType(typeof(OperationOrResponseInternals<>), operation.ResultType) : new CSharpType(typeof(OperationOrResponseInternals));
+                var waitForCompleteMethodName = operation.ResultType != null ? "WaitForCompletionAsync" : "WaitForCompletionResponseAsync";
+
+                using (writer.Scope($"{operation.Declaration.Accessibility} partial class {cs.Name}: {baseType}"))
+                {
+                    writer.Line($"private readonly {helperType} _operation;");
+
+                    writer.Line();
+                    writer.WriteXmlDocumentationSummary($"Initializes a new instance of {cs.Name} for mocking.");
+                    using (writer.Scope($"protected {cs.Name:D}()"))
+                    {
+                    }
+
+                    writer.Line();
+                    writer.Append($"internal {cs.Name}(");
+                    // todo: programmatically get the type of operationBase from the definition of [Resource]
+                    writer.Append($"{typeof(ResourceOperationsBase)} operationsBase, ");
```

Comment:
I hard coded `ResourceOperationsBase` because there are no info exactly which type should be used in the resource's constructor, and in `ResourceWriter` it's hard coded as well.
https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs#L31

Reply:
Ok, I will address it in the Singleton work item. Let's leave it as hardcoded for now.